Shared/ViewModels/ItemViewModel/SeriesItemViewModel.swift

```diff
@@ -131,11 +131,13 @@ final class SeriesItemViewModel: ItemViewModel, MenuPosterHStackModel {
             let request = Paths.getSeasons(seriesID: item.id!, parameters: parameters)
             let response = try await userSession.client.send(request)
 
-            guard let seasons = response.value.items else { return }
-
+            let seasons = response.value.items ?? []
             if let firstSeason = seasons.first {
                 self.getEpisodesForSeason(firstSeason)
                 await MainActor.run {
+                    for season in seasons {
```

Comment:
Move this instead to right below checking whether we have seasons, like in: https://github.com/jellyfin/Swiftfin/pull/883/files#diff-1462c1160e5edfcfdbf96fa2622aeaead21bbc319987b8bfb6ace9cd32e66382L143

Reply:
I am not super familiar with MainActor, but if implemented as before doesn't this task wait for the dispatch to the main thread to execute before continuing execution?  If so this wouldn't that introduce latency.

Reply:
That little latency is fine. In fact, `getEpisodesForSeason` should be migrated to `async` so we would have to `await` that before setting the seasons at all with this<sup>1</sup>. Instead, we would want a loading state while retrieving the current season's episodes. This is what I did (very badly) before #883 and in #905 (along with the foundation for lazily loading) this will be done again.

In fact _again_ we should actually probably create the empty-state dictionary and then set it so that the view updates only once in response to the `Publisher` emitting a change but 🤷 

---

1 - The basic `Task` wrapped function bodies were just to make the old `Publisher` -> `async/await` transition feasible.